src/utils/getActualDataList.ts

```diff
@@ -0,0 +1,31 @@
+import dayjs, { extend } from 'dayjs';
+import customParseFormat from 'dayjs/plugin/customParseFormat';
+
+type DataListType<T> = Array<T>;
+type DataItem = Record<string, string | HTMLElement>;
+interface DataItemWithDate extends DataItem {
+  date: string; // 'YYYY-MM-DD'
+}
+
+export type ActualDataListProps = {
+  dataList: DataListType<DataItemWithDate>;
+  actualDelayInDays: number;
+  sorted?: boolean;
+};
+
+export const getActualDataList = ({
+  dataList,
+  actualDelayInDays,
+  sorted = true,
+}: ActualDataListProps): DataListType<DataItemWithDate> | [] => {
+  extend(customParseFormat);
+
+  const postponeDate = dayjs().subtract(actualDelayInDays, 'day');
+  const actualDataList = dataList.filter((item) => dayjs(item.date, 'YYYY-MM-DD') >= postponeDate);
+
+  const sortedList = (actualDataList as DataListType<DataItemWithDate>).sort((a, b) =>
```

Comment:
```suggestion
  const sortedList = actualDataList.sort((a, b) =>
```

Reply:
Since I'm sorting by `date` field, I need to indicate for TS that the sorted items definitely have a `date` field

Reply:
TypeScript is smart enough to understand that without type casting

![image](https://github.com/rolling-scopes/site/assets/114234698/fcc345df-99d7-4378-aa03-a0899435c769)